Sources/Xunit.Autofac/Execution/AutofacTestCollectionRunner.cs

```diff
@@ -18,7 +18,9 @@ internal class AutofacTestCollectionRunner: XunitTestCollectionRunner
             ILifetimeScope lifetimeScope
             ) : base(testCollection, testCases, diagnosticMessageSink, messageBus, testCaseOrderer, aggregator, cancellationTokenSource)
         {
-            _lifetimeScope = lifetimeScope;
+            var nestedScope = lifetimeScope.BeginLifetimeScope(LifetimeScopeTags.TestCollection);
```

Comment:
Why are you creating an explicit lifetime scope instead of using the parent one, created in `AutofacTestAssemblyRunner.RunTestCollectionAsync` exclusively for running this test collection? You can easily mark it with your tag as well.